### PR TITLE
Bug fix + test - Sequential.pop()

### PR DESCRIPTION
**Pull request by farizrahman4u**
Update inbound nodes after popping last layer. Otherwise `output_shape` of the model will not be updated.

```python
model = Sequential()
model.add(Dense(10, input_dim=10))
model.add(Dense(20))
print model.output_shape  # >> (None, 20)
model.pop()
print model.output_shape  # >> (None, 20) 
```

Also since `pop()` is sort of a hack, I thought it would be good to have a test.

### CODE REVIEW

**Comment by fchollet** [comment]
LGTM, thanks. Will wait for the tests to complete before merging.

**Comment by alyato** [comment]
@farizrahman4u 
  I wanna use this pretrained vgg19.  

>  x = Flatten(name='flatten')(x)
        x = Dense(4096, activation='relu', name='fc1')(x)
        x = Dense(4096, activation='relu', name='fc2')(x)
        x = Dense(1000, activation='softmax', name='predictions')(x)
 model = Model(img_input, x)
    return model

 i have only 8 labels to classlify. 
 But it is used by the keras function API ,not the `Sequential` model.
 how can i use the `model.layers.pop()`
like this

>  x = Dense(1000, activation='softmax', name='predictions')(x)
model = Model(img_input, x)
model.layers.pop()
 x = Dense(8, activation='softmax', name='predictions')(x)
 loading weight
return model

Do you give me some advices? Thanks
